handlers/xpHandler.go

```diff
@@ -0,0 +1,141 @@
+package handlers
+
+import (
+	"math"
+	"math/rand"
+	"strings"
+	"time"
+
+	controllers "tpc-discord-bot/controllers/levels"
+
+	"github.com/bwmarrin/discordgo"
+)
+
+type LevelingConfig struct {
+	XpRate float64
+}
+
+func HandleXpGive(s *discordgo.Session, m *discordgo.MessageCreate) {
+	// Skip if message is from bot or not in guild
+	if m.Author.Bot || m.GuildID == "" {
+		return
+	}
+
+	controller := &controllers.LeaderboardController{}
+	levelingConfig := LevelingConfig{XpRate: 1.0}
+
+	// List of XP-enabled channels
+	xpChannels := []string{
+		"crew-chat", "fly-with-me", "Event Comms", "Flight Voice 1",
+		"Flight Voice 2", "Shared Cockpit", "Magneto Lounge", "1 on 1",
+		"general-aviation-ops", "streams-and-videos", "screenshots",
+		"Charters Comms", "community-help-forum", "pilot-training-chat",
+		"charters-chat", "explorer-missions", "irl-pilots", "atc-lounge",
+		"company-perks", "early-adopters", "airliner-and-bizjet-ops",
+		"rotary-ops", "your-setups",
+	}
+
+	// Get channel
+	channel, err := s.Channel(m.ChannelID)
+	if err != nil {
+		return
+	}
+
+	// Check if channel is in XP channels list
+	channelAllowed := false
+	for _, xpChannel := range xpChannels {
+		if strings.EqualFold(channel.Name, xpChannel) {
+			channelAllowed = true
+			break
+		}
+	}
+	if !channelAllowed {
+		return
+	}
+
+	// Find or create user
+	user, err := controller.FindUser(m.Author.ID)
+	if err != nil {
+		// Add user if not found
+		err = controller.AddUser(m.Author.ID, m.GuildID)
+		if err != nil {
+			return
+		}
+		return
```

Comment:
Returning at this point won't have it come back and finish the function (giving points). Is this the expected outcome?

Reply:
I feel like we should still give them some points even on first message as I think thats the current config. Can just update it to have the point values added